### ISSUE TITLE
NPM fails to install @pnpm/bundled

### ISSUE THREAD

**Issue by pgonzal** [pnpm/pnpm#1032]
### Code to reproduce the issue:

```
C:\>md repro
C:\repro>cd repro
C:\repro>npm init
C:\repro>npm install @pnpm/bundled@1.31.6
```

### Expected behavior:

It should install successfully.

### Actual behavior:

It fails to install, with this error:
```
18822 silly saveTree   |   `-- write-file-atomic@1.3.4
18822 silly saveTree   |     +-- imurmurhash@0.1.4
18822 silly saveTree   |     `-- slide@1.1.6
18822 silly saveTree   +-- text-table@0.2.0
18822 silly saveTree   +-- tree-kill@1.2.0
18822 silly saveTree   +-- update-notifier@2.3.0
18822 silly saveTree   +-- util.promisify@1.0.0
18822 silly saveTree   `-- write-json-file@2.3.0
18823 warn repro@1.0.0 No description
18824 warn repro@1.0.0 No repository field.
18825 verbose stack Error: EPERM: operation not permitted, rename 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\.glob.DELETE\node_modules\@types' -> 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\glob\node_modules\@types'
18825 verbose stack     at Error (native)
18826 verbose cwd D:\repro
18827 verbose Windows_NT 10.0.16299
18828 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "@pnpm/bundled"
18829 verbose node v6.12.3
18830 verbose npm  v5.6.0
18831 error path D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\.glob.DELETE\node_modules\@types
18832 error code EPERM
18833 error errno -4048
18834 error syscall rename
18835 error Error: EPERM: operation not permitted, rename 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\.glob.DELETE\node_modules\@types' -> 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\glob\node_modules\@types'
18835 error     at Error (native)
18835 error  { Error: EPERM: operation not permitted, rename 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\.glob.DELETE\node_modules\@types' -> 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\glob\node_modules\@types'
18835 error     at Error (native)
18835 error   cause:
18835 error    { Error: EPERM: operation not permitted, rename 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\.glob.DELETE\node_modules\@types' -> 'D:\repro\node_modules\@pnpm\bundled\node_modules\@pnpm\default-fetcher\node_modules\@pnpm\tarball-fetcher\node_modules\@types\rimraf\node_modules\@types\glob\node_modules\@types'
18835 error        at Error (native)
18835 error      errno: -4048,
18835 error      code: 'EPERM',
18835 error      syscall: 'rename',
18835 error      path: 'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\.glob.DELETE\\node_modules\\@types',
18835 error      dest: 'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\glob\\node_modules\\@types' },
18835 error   stack: 'Error: EPERM: operation not permitted, rename \'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\.glob.DELETE\\node_modules\\@types\' -> \'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\glob\\node_modules\\@types\'\n    at Error (native)',
18835 error   errno: -4048,
18835 error   code: 'EPERM',
18835 error   syscall: 'rename',
18835 error   path: 'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\.glob.DELETE\\node_modules\\@types',
18835 error   dest: 'D:\\repro\\node_modules\\@pnpm\\bundled\\node_modules\\@pnpm\\default-fetcher\\node_modules\\@pnpm\\tarball-fetcher\\node_modules\\@types\\rimraf\\node_modules\\@types\\glob\\node_modules\\@types',
18835 error   parent: '@types/rimraf' }
18836 error Please try running this command again as root/Administrator.
18837 verbose exit [ -4048, true ]
```

### Additional information:

 - `node -v` prints: 6.12.3
 - Windows, OS X, or Linux?: Windows

Note that `yarn add @pnpm/bundled` and `pnpm install @pnpm/bundled` both work fine.  We tried several other versions of NPM and encountered similar errors.

Perhaps this is an NPM bug or limitation.  However, since we're using this to bootstrap PNPM in an automated environment, we really need it to work with NPM.  It would be weird to require a full install of Yarn or regular PNPM before we can install @pnpm/bundled.

**Comment by pgonzal** [comment]
@nickpape-msft FYI

**Comment by pgonzal** [comment]
I also confirmed that this issue is not specific to a particular release of `@pnpm/bundled`.  I get the same failure for `@pnpm/bundled@1.20.0` and `@pnpm/bundled@1.32.0`

**Comment by zkochan** [comment]
I don't understand why npm tries to rename something inside node_modules. Everything in node_modules is bundled, so npm should not touch it.

We used to have a different approach. We moved node_modules to `lib/node_modules` and renamed the `dependencies` field in package.json. During publish npm ignores only the node_modules in the root of the package. So a node_modules in a subfolder gets published. That solution stopped working for some reason but I think if we want it to work reliably with npm, we should revisit it. By the way, that's how bower is published as well.

Yarn has a different approach. It bundles its sources with webpack. That might be a good solution as well

The one thing that makes it a bit harder is that pnpm has a native dependency: [drivelist](https://github.com/resin-io-modules/drivelist)

**Comment by pgonzal** [comment]
Maybe you could take a direct dependency on **drivelist** using a specific version number?

**Comment by pgonzal** [comment]
> Yarn has a different approach. It bundles its sources with webpack. That might be a good solution as well

@iclanton FYI... by coincidence we were were talking about webpacking tools today and wondering if it might be faster.

**Comment by zkochan** [comment]
Fixed in [pnpm@1.33.2]. pnpm is now bundled with its dependencies using [publish-packed](https://github.com/zkochan/publish-packed)
